Sources/TestingMacros/PragmaMacro.swift

```diff
@@ -0,0 +1,79 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+public import SwiftSyntax
+public import SwiftSyntaxMacros
+
+/// A type describing the expansion of the `@__testing` attribute macro.
+///
+/// Supported uses:
+///
+/// - `@__testing(semantics: "nomacrowarnings")`: suppress warning diagnostics
+///   generated by macros. (The implementation of this use case is held in trust
+///   at ``MacroExpansionContext/areWarningsSuppressed``.
+///
+/// This type is used to implement the `@__testing` attribute macro. Do not use
+/// it directly.
+public struct PragmaMacro: PeerMacro, Sendable {
+  public static func expansion(
+    of node: AttributeSyntax,
+    providingPeersOf declaration: some DeclSyntaxProtocol,
+    in context: some MacroExpansionContext
+  ) throws -> [DeclSyntax] {
+    return []
+  }
+
+  public static var formatMode: FormatMode {
+    .disabled
+  }
+}
+
+/// Get all pragma attributes (`@__testing`) associated with a syntax node.
+///
+/// - Parameters:
+///   - node: The syntax node to inspect.
+///
+/// - Returns: The set of pragma attributes strings associated with `node`.
+///
+/// Attributes conditionally applied with `#if` are ignored.
+func pragmas(on node: some WithAttributesSyntax) -> [AttributeSyntax] {
+  node.attributes
+    .compactMap { attribute in
+      if case let .attribute(attribute) = attribute {
+        return attribute
+      }
+      return nil
+    }.filter { attribute in
+      attribute.attributeNameText == "__testing"
```

Comment:
Is this robust enough to accept a fully-qualified name with module prefix like `@Testing.__testing` ?

Reply:
(And conversely, to reject attributes with the same name which could hypothetically exist in other modules?)

Reply:
FBOFW that isn't actually supported by the language, so… no. But I can code it defensively if you're concerned.

Reply:
Actually, no, I don't think we care: realistically nobody else is going to write `@__testing(semantics: "...")`. If somebody really _does_ run into trouble here, we can address it in the future.

Reply:
Added fully-qualified name check. If somebody else makes a macro named `@__testing(semantics:)`, it'll trip as ambiguous before getting to our code, so I don't care too much. :)